haskell/test/Test/Ligo/BaseDAO/Integrational/Common.hs

```diff
@@ -10,18 +10,28 @@ import Universum
 import Lorentz
 import Lorentz.Test
 
+import Data.List.NonEmpty as NE (toList)
 import qualified Ligo.BaseDAO.Contract as DAO
 import Ligo.BaseDAO.Types
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
 import Michelson.Test.Integrational (tOriginate)
-import Data.List.NonEmpty as NE (toList)
+import Test.Ligo.BaseDAO.Common (dummyFA2Contract)
 
 withOriginated
   :: Int
   -> ([Address] -> FullStorage)
   -> ([Address] -> TAddress Parameter -> IntegrationalScenarioM a)
   -> IntegrationalScenarioM a
 withOriginated addrCount storageFn tests = do
+  tokenContract <- lOriginate dummyFA2Contract "TokenContract" [] (toMutez 0)
   let addresses = take addrCount $ NE.toList $ genesisAddresses
-  let storage = storageFn addresses
+  let storageInitial = storageFn addresses
+  let storage = storageInitial { fsStorage = (fsStorage storageInitial)
+    { sGovernanceToken = GovernanceToken
+        { gtAddress = unTAddress tokenContract
+        , gtTokenId = FA2.TokenId 0
+        }
+    } }
```

Comment:
Why not have `storageFn` take care of changing the `GovernanceToken` as well?

Reply:
`storageFn` was pure as of now, so couldn't orignate the FA2 contract. Also the need for it didn't arise yet. But I have changed it now anyway, since the change required was small.